packages/@lwc/integration-karma/test/rendering/callback-invocation-order/index.spec.js

```diff
@@ -297,3 +299,90 @@ it('should invoke callbacks on the right order when multiple templates are used
             ]);
         });
 });
+
+if (process.env.NATIVE_SHADOW) {
```

Comment:
Rather than run this test only in native shadow, I would prefer we run it in both and validate the behavior difference between the two. We can still reference the bug in a comment.

Reply:
I updated in the latest commit to run both native and synthetic shadow.